frontend/src/components/associations/events/list_events/ListEvents.tsx

```diff
@@ -0,0 +1,142 @@
+
+import { Link } from "react-router-dom";
+import { api } from "../../../../services/apiService";
+import Card from "react-bootstrap/Card";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Container from "react-bootstrap/Container";
+import { LoadingAssociation } from "../../Loading";
+import { PageTitle } from "../../../utils/PageTitle";
+import { Pagination } from "../../../utils/Pagination";
+import { Association } from "../../../../models/associations/association";
+import { EventCard } from "./EventCard";
+import { UserContext } from "../../../../services/authService";
+import { SidebarOption } from "../../../utils/sidebar/interfaces";
+
+export const AssociationEventList = ({
+    association,
+    setSidebar,
+}: {
+    association: Association;
+    setSidebar: (
+        value:
+            | ((prevState: SidebarOption | null) => SidebarOption | null)
+            | SidebarOption
+            | null
+    ) => void;
+}) => {
+    const user = useContext(UserContext);
+    const [parameters, setParameters] = useState<{
+        association: string;
+        time: ("NOW" | "BEFORE" | "AFTER")[];
+    }>({
+        association: association.id,
+        time: [],
+    });
+
+    useEffect(() => {
+        setSidebar({
+            notifyChange: (sidebarParameters) => {
+                const time: ("NOW" | "BEFORE" | "AFTER")[] = [];
+
+                if (sidebarParameters["date.before"]) {
+                    time.push("BEFORE");
+                }
+
+                if (sidebarParameters["date.now"]) {
+                    time.push("NOW");
+                }
+
+                if (sidebarParameters["date.after"]) {
+                    time.push("AFTER");
+                }
+
+                setParameters((oldParameters) => ({
+                    ...oldParameters,
+                    time: time,
+                }));
+            },
+            searchable: false,
+            sections: [
+                {
+                    title: "Voir les événements…",
+                    id: "date",
+                    retractable: false,
+                    fields: [
+                        {
+                            type: "checkbox",
+                            id: "before",
+                            label: "Terminés",
+                            defaultValue: false,
+                        },
+                        {
+                            type: "checkbox",
+                            id: "now",
+                            label: "En cours",
+                            defaultValue: true,
+                        },
+                        {
+                            type: "checkbox",
+                            id: "after",
+                            label: "À venir",
+                            defaultValue: true,
+                        },
+                    ],
+                },
+            ],
+        });
+    }, [setParameters, setSidebar]);
```

Comment:
Options sidebar is being refactored, we'll need to wait for it to be merged to merge #135